front/src/applications/operationalStudies/components/Scenario/AddOrEditScenarioModal.tsx

```diff
@@ -105,48 +119,58 @@ export default function AddOrEditScenarioModal({
   const createScenario = async () => {
     if (!currentScenario.name || !currentScenario.infra_id) {
       setDisplayErrors(true);
-    } else {
-      try {
-        const result = await post(`${rootURI}`, { ...currentScenario, infra_id: infraID });
-        dispatch(updateScenarioID(result.id));
-        navigate('/operational-studies/scenario');
-        closeModal();
-      } catch (error) {
-        console.error(error);
-      }
+    } else if (infraID && projectID && studyID && currentScenario && infraID) {
+      postScenario({
+        projectId: projectID,
+        studyId: studyID,
+        scenarioRequest: { ...(currentScenario as ScenarioRequest), infra_id: infraID },
+      })
+        .unwrap()
+        .then(({ id }) => {
+          dispatch(updateScenarioID(id));
+          navigate('/operational-studies/scenario');
+          closeModal();
+        })
+
+        .catch((error) => console.error(error));
```

Comment:
It would be nice to show a notification to the user here if the request fails

Reply:
Done !!!